tests/testsuite/directory.rs

```diff
@@ -95,6 +95,30 @@ fn truncated_directory_in_root() -> io::Result<()> {
     Ok(())
 }
 
+#[test]
+#[ignore]
+fn truncated_directory_config() -> io::Result<()> {
```

Comment:
Looks good! 👍 
This covers the case where the directory isn't truncated because its length is so long.
Could we add one extra test here to ensure that truncation does happen correctly after it has been changed? (e.g. `directory.truncation_length = 2`)